framework/src/meshmodifiers/MeshExtruder.C

```diff
@@ -39,14 +39,23 @@ InputParameters validParams<MeshExtruder>()
 
 MeshExtruder::MeshExtruder(const InputParameters & parameters) :
     MeshModifier(parameters),
-    _num_layers(getParam<unsigned int>("num_layers")),
     _extrusion_vector(getParam<RealVectorValue>("extrusion_vector")),
+    _num_layers(getParam<unsigned int>("num_layers")),
     _map_custom_ids(false)
 {
   if (isParamValid("existing_subdomains") || isParamValid("layers") || isParamValid("new_ids"))
   {
     if (isParamValid("existing_subdomains") && isParamValid("layers") && isParamValid("new_ids"))
+    {
       _map_custom_ids = true;
+      _existing_subdomains = getParam<std::vector<SubdomainID> >("existing_subdomains");
+      _layers =  getParam<std::vector<unsigned int> >("layers");
+      _new_ids = getParam<std::vector<unsigned int> >("new_ids");
```

Comment:
yes. We can check if the size of `existing_subdomains` is zero or not instead of setting `_map_custom_ids`.